akka-persistence/src/main/scala/akka/persistence/Eventsourced.scala

```diff
@@ -384,6 +385,12 @@ private[persistence] trait Eventsourced extends Snapshotter with PersistenceStas
    */
   def recoveryFinished: Boolean = !recoveryRunning
 
+  override def stash(): Unit = {
+    if (withinHandler)
+      throw new IllegalStateException("Do not call stash inside of persist callback.")
```

Comment:
Perhaps we could go into some more detail on why this is bad and what to do instead?

Reply:
The current message that is to be stashed is an internal message from the journal confirming the persist (`WriteMessageSuccess`). Stashing that is obviously not desired.

Reply:
We can solve it without the flag. We can check if the currentMessage is
```
context.asInstanceOf[ActorCell].currentMessage match {
  case _: JournalProtocol.Response => throw new IllegalStateException(...
  case _ => super.stash()
}
```

This also has the benefit that we cover some more cases, such as messages during recovery.

Reply:
@ortigali here is the suggestion ^

Reply:
@ortigali Will you be able to complete this with the approach suggested, or should we take over from here?